File: backend/controllers/user.controller.js
const userRepository = require('../repositories/user.repository');

const getProfile = async (req, res) => {
  try {
    // req.user je postavljen od strane authMiddleware-a
    const userId = req.user.id; 
    const user = await userRepository.findUserById(userId);
    if (!user) {
      return res.status(404).json({ message: "Korisnik nije pronađen." });
    }
    res.json(user);
  } catch (error) {
    res.status(500).json({ message: "Greška na serveru." });
  }
};

module.exports = { getProfile };

File: backend/controllers/user_controller.go
package controllers

import (
	"net/http"

	"github.com/Majlo123/sugar-beet-backend/repositories"
	"github.com/Majlo123/sugar-beet-backend/services"
	"github.com/gin-gonic/gin"
)

func GetProfile(c *gin.Context) {
	val, exists := c.Get("user")
	if !exists {
		c.JSON(http.StatusInternalServerError, gin.H{"message": "Server error."})
		return
	}
	claims, ok := val.(*services.CustomClaims)
	if !ok {
		c.JSON(http.StatusInternalServerError, gin.H{"message": "Server error."})
		return
	}

	user, err := repositories.FindUserByID(claims.ID)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"message": "Server error."})
		return
	}
	if user == nil {
		c.JSON(http.StatusNotFound, gin.H{"message": "User not found."})
		return
	}

	c.JSON(http.StatusOK, user)
}
